src/exporter.js

```diff
@@ -133,9 +136,22 @@ export default class {
   _tagsToStr(tags=[]){
     return tags.map(tag => tag.replace(/ /g, '_')).join(' ');
   }
-}
 
-export const rand = () => Math.random() * 100000000 | 0;
+  _getId(table, col, ts) {
+    const query = `SELECT ${col} from ${table} WHERE ${col} >= ${ts}`;
```

Comment:
> yep. But then we'll lost logic about datetime. Using max value was a continue of my question about non-random ids ( as result of using autoincrement ).

We always insert either current date or (current date + n) where n is very small

> Currently I'll reimplement original logic by link you provide in issues discussion 

My response was based on your initial code 

```js
    const query = `SELECT ${col} from ${table} WHERE ${col} >= ${ts}`;
    const queryResult = this.db.exec(query);
    if (queryResult.length === 0) {
      return ts;
    }
    const [ { values } ] = queryResult;
    const existingValues = values.map(([colValue]) => +colValue);
    while (existingValues.indexOf(ts) > -1) {
      ts++;
    }
```

I really didn't like the part where you had `while` and `indexOf`. 

When you are checking for containing in the returned array, you have two options - either the array have a hole (1 2 4, you are inserting value 3) or the values is consequent (1 2 3, you are inserting value 4).

But in the first case, you will break the "rule" that cards IDs should be in order (first added - lower id)

Now the second case, you need to find only the maximum value inside the array and increment it. Instead of doing it in JavaScript, you can use `SQL`'s `ORDER BY`